shortfin/python/shortfin_apps/llm/components/request_queue_manager.py

```diff
@@ -80,3 +91,37 @@ def remove_from_queue(self, id: Optional[int]) -> None:
             logger.debug(
                 f"Removed from queue: new queue size {self._current_queue_size}"
             )
+
+    def check_memory_availability(
```

Comment:
We should be adding this behavior to `add_to_queue`. Any of the rate limiter work should check all resources rather than individually checking each. The idea is we can ask the manager whether we have all resources available to run. Having separate checks and returns just clutters up the python code.

So you will need to augment `add_to_queue` to take any other available work.